compliance_checker/ioos.py

```diff
@@ -888,89 +889,163 @@ def check_platform_vocabulary(self, ds):
-        denoting "true" or "false".
-
-        Any variables which a user would like ingested must also contain the
-        gts_ingest attribute with a value of true. The variable must:
+        the gts_ingest is "required, if applicable". Any variables which a user
+        would like ingested must also contain the gts_ingest attribute with a
+        value of true. The variable must:
           - have a valid CF standard_name attribute (already checked)
           - have an ancillary variable reqpresenting QARTOD aggregate flags
           - have a valid udunits units attribute (already checked)
 
-        Args:
-            ds (netCDF4.Dataset): open Dataset
+        Parameters
+        ----------
+        ds (netCDF4.Dataset): open Dataset
 
-        Returns:
-            list of Result objects
+        Returns
+        -------
+        Result
         """
 
-        default_pass_result = Result(BaseCheck.HIGH, True, "gts_ingest", ["gts_ingest"])
-
-        results = []
-
-        # check global
-        glb_msg = ("If provided, the global attribute \"gts_ingest\" must be a "
-                   "string and its value must be one of \"true\" or \"false\"")
+        # is dataset properly flagged for ingest?
         glb_gts_attr = getattr(ds, "gts_ingest", None)
         if glb_gts_attr and (glb_gts_attr=="true" or glb_gts_attr=="false"):
             do_gts = True if glb_gts_attr == "true" else False
-            results.append(Result(BaseCheck.HIGH, True, "gts_ingest", [glb_msg]))
         else:
             do_gts = False
-            results.append(default_pass_result)
 
         # check variables
-        var_msg = ("The attribute \"gts_ingest\" of variable \"{v.name}\" "
-                   " must fulfill the following:\n"
-                   "  - must be a string with value \"true\" or \"false\";\n"
-                   "  - have a valid CF standard_name attribute (already checked);\n"
-                   "  - have an ancillary variable representing QARTOD aggregate flags;\n"
-                   "  - have a valid udunits units attribute\n"
-                   "The global attribute \"gts_ingest\" "
-                   "must also have a value of \"true\".")
-
-        for v in ds.get_variables_by_attributes(gts_ingest=lambda x: x=="true"):
-            results.append(self._check_var_gts_ingest(ds, v, do_gts, var_msg.format(v=v)))
+        all_passed_ingest_reqs = True # default
 
-        return results
+        if do_gts: # execute if dataset flagged for ingest
+
+            var_passed_ingest_reqs = set()
+            var_passed_ingest_msg  = "The following variables qualifed for GTS Ingest:"
+            var_failed_ingest_msg  = "The following variables did not qualify for GTS Ingest:"
+
+            for v in ds.get_variables_by_attributes(gts_ingest=lambda x: x=="true"):
+                var_passed_ingest_reqs.add((v.name, self._var_qualifies_for_gts_ingest(ds, v)))
+
+            for vtup in var_passed_ingest_reqs:
```

Comment:
Use tuple unpacking for the arguments here with descriptive variable names.  I'm not sure what the elements of `vtup` are supposed to represent.

Reply:
Good call here, descriptive variable names are always useful.